src/harmonic.jl

```diff
@@ -0,0 +1,48 @@
+"""
+    harmonic(n)
+
+ Calculates harmonic numbers
+   e.g., see http://mathworld.wolfram.com/HarmonicNumber.html
+
+## Arguments
+* `n::Integer`: index of the Harmonic number to calculate
+
+## Examples
+```jldoctest
+julia> harmonic(2)
+1.5
+```
+"""
+function harmonic(n::Integer)
+    # γ = euler_mascheroni_const = 0.577215664901532860606512090082402431042 # http://oeis.org/A001620
+    if n <=10
+        # get exact values for small n
+        return sum( 1.0./(1:n) )
```

Comment:
Write out loops, don't use `sum` of a `broadcast`.  This isn't Matlab.

See [Why vectorized code is not as fast as it could be](https://julialang.org/blog/2017/01/moredots) and [here](http://www.johnmyleswhite.com/notebook/2013/12/22/the-relationship-between-vectorized-and-devectorized-code/) for example.   You are allocating a temporary array and then looping over it with `sum`.